parserator_web/views.py

```diff
@@ -16,9 +15,30 @@ class AddressParse(APIView):
     def get(self, request):
         # TODO: Flesh out this method to parse an address string using the
         # parse() method and return the parsed components to the frontend.
-        return Response({})
+
+        address_components, address_type = {}, ''
+
+        # in case the this endpoint is ever called without the
+        # address parameter
+        if 'address' not in request.query_params:
+            return Response({'input_string': '',
+                             'address_components': address_components,
+                             'address_type': address_type})
+
+        input_string = request.query_params['address']
+
+        try:
+            address_components, address_type = self.parse(input_string)
+        except usaddress.RepeatedLabelError as e:
+            address_components, address_type = e.parsed_string, 'Error'
+        finally:
+            return Response({'input_string': input_string,
+                             'address_components': address_components,
+                             'address_type': address_type})
```

Comment:
The code inside your `finally` block will execute unconditionally. In effect, this means any unhandled errors (e.g., if you accidentally forgot to define the `input_string`, resulting in a `NameError` on L31) will get swallowed, leading to difficult debugging and confusion for the user, who will receive an empty response despite providing an address. 

I'd suggest returning your `Response` outside of the `try`/ `except` block so broken code will fail helpfully.

```suggestion
        return Response({'input_string': input_string,
                         'address_components': address_components,
                         'address_type': address_type})
```

Reply:
Great suggestion. Removing the `finally` block will definitely make future debugging easier.